doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+      \inode[below = 40 pt]{b}{$⊗$}
+      \vwire{a}{b}
+    \end{tikzpicture}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.05\textwidth}
+    \huge{→}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+  \end{subfigure}
 \end{figure}
 
 \floatstyle{boxed}
 \restylefloat{figure}
 
 ### Argument for correctness
 
+
 ## Bespoke term encoding
 
 - Primitive types (integer, string, bytes) ~> node types w/data
 - n-arity primitive functions (+, -, *, concat, index) ~> rewrite rules
-- alt-term encoding: $lam x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
+- alt-term encoding: $\lambda x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
 
 - Over set of underlying opcodes & von-Neumann-ish VM (e.g. x86, LLVM, EVM).
 - Use linear types (e.g. rewrite memory locations in place)
 - Must be semantically equivalent to interaction system encoding after read-back
 - Must preserve any-order reduction (exists some maximum `n` rewrite steps after which graph read-back must have final form regardless of order)
 - Can relax diamond property (strong confluence)? unclear if necessary
 
+### Argument for Correct read-back
+
+We wish to show that a read-back of an EAL-expression that has undergone zero or more reductions gives back the same result as normal evaluation up to alpha equality.
+
+Before we can prove this result, we must first prove a few lemmas and theorems first as well talk about what fan in's with the same label imply.
+
+The first lemma we wish to prove is the following
+
+\begin{Lemma}{AST→Net has one free port}{AST->Net}
+  Let $A$ = be a valid BOHM ADT.
+  \\ \\
+  Now Consider the net over $A$, let $N_a$ be this net.
+  \\ \\
+  Now let us consider any sub case of $A$, $L = P(n_1 \star \cdots \star n_k)$.
+  \\ \\
+  The node corresponding to $L$ looks.
+
+  \begin{tikzpicture}
+    \draw (0,0) circle (.8cm);
+  \end{tikzpicture}
+
+  Where the labels in $N_a$ are the same as in $L$, with $n_{k+1}$ being an extra port which connects to the ADT above it.
+  \\ \\
+  For an ADT to be realized, we must have all arguments, so all these ports must be connected.
+  Now, if $L$ is $A$, then we have one free port, however if it is not, then this free port must not be free but instead be connected to the ADT above it.
+  \\ \\
+  For free variables which have no $\lambda$ to connect to, a symbol node is created, maintaining the invariant.
+  \\ \\
+  $\therefore$ AST→Net has one free port
+\end{Lemma}
+
+
+Now we need to prove that evaluating this net does not change this fact
+
+
+\begin{Theorem}{AST→Net→Eval has one free port}{AST->Net->Eval}
+  Let $A = Net(AST)$.
+  \\ \\
+  By lemma \ref{Lm:AST->Net} $A$ has one free port.
+  \\ \\
+  We must now show $Eval(A)$ does not change the number of free ports.
+  \\ \\
+  We can show this by simply considering how every reduction rule works. For the sake of brevity, we will only consider the three variations of rewrite rules present in the BOHM system WLOG.
```

Comment:
What is WLOG?

I think you mean that every rule takes one of these forms with `f` and `g` varying, but that's not clear.

I don't think it's true unless you allow the number of auxiliary ports to vary.

Reply:
>What is WLOG?

without loss of generality. It's often used in proofs where one doesn't want to specify every case, as the cases they cover are general and cover all cases by using a specific case

> I think you mean that every rule takes one of these forms with f and g varying, but that's not clear.

Yeah, I could  have been clearer.

> I don't think it's true unless you allow the number of auxiliary ports to vary.

Ideally they would be, but I couldn't figure out how to add ...'s to the diagram, so I left them with 2 arguments

Reply:
> without loss of generality. It's often used in proofs where one doesn't want to specify every case, as the cases they cover are general and cover all cases by using a specific case

OK we should define it somewhere, it can be earlier.

> Ideally they would be, but I couldn't figure out how to add ...'s to the diagram, so I left them with 2 arguments

Alright, we should mention it, even as a TODO.

Reply:
I have added todo's now.

The first point still isn't addressed